### DISCUSSION TITLE
iOS build taking infinite time to build - Previously it was working fine with the same yaml configuration.

### DISCUSSION THREAD

**Discussion by yasirali133314** [codemagic-ci-cd/codemagic-docs#2793] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Here is my codemagic iOS workflow.

  ios-workflow:
    name: IOS Workflow
    max_build_duration: 120
    instance_type: mac_mini_m1
    integrations:
      app_store_connect: Codemagic Store Key
    environment:
      ios_signing:
        distribution_type: app_store
        bundle_identifier: com.app.myApp
      vars:
        APP_STORE_APPLE_ID: my_apple_id_here
      flutter: stable
      xcode: latest
      cocoapods: default
    scripts:
      - name: Set up code signing settings on Xcode project
        script: |
          xcode-project use-profiles
      - name: Get Flutter packages
        script: |
          flutter packages pub get
      - name: Install pods
        script: |
          find . -name "Podfile" -execdir pod install \;
      - name: Flutter build ipa and automatic versioning
        script: |
          flutter build ipa --release \
            --export-options-plist=/Users/builder/export_options.plist
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - flutter_drive.log
    publishing:
      email:
        recipients:
          - yasirali8889905@gmail.com
        notify:
          success: true
          failure: true
      app_store_connect:
        auth: integration
        submit_to_testflight: true
        submit_to_app_store: false

### Expected results

It should submit build to the testflight.

### Actual results

Infinite loading on building IPA

**Comment by dam-ease** [comment]
Hello @yasirali133314,
Please can you share your buildID with me? You can find it under your build details page's 'ID' section. ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Reply by yasirali133314** [reply]
@dam-ease here is my recent build ID for iOS: 67177ef8462ddc9c081e88ee

**Reply by dam-ease** [reply]
Hello @yasirali133314.
I looked through the build logs and can see this error in the pod install step:
```
[!] Invalid `Podfile` file: /Users/builder/clone/ios/Flutter/Generated.xcconfig must exist. If you're running pod install manually, make sure flutter pub get is executed first.

 #  from /Users/builder/clone/ios/Podfile:19
 #  -------------------------------------------
 #    unless File.exist?(generated_xcode_build_settings_path)
 >      raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
 #    end
 #  -------------------------------------------
```

Can you please add the below script to your yaml file, 
```
cd ios
rm -rf Podfile.lock
flutter clean
flutter pub get
pod install
pod repo update
```
Also remove the backslash from your build command i.e you have:
```
flutter build ipa --release 
  --export-options-plist=/Users/builder/export_options.plist
```
And rerun the build. Please do let me know how this goes on your end.

**Reply by yasirali133314** [reply]
@dam-ease I tried, it is taking same amount of time and never finish the iOS build.

here is my updated yaml file.

  ios-workflow:
    name: IOS Workflow
    max_build_duration: 120
    instance_type: mac_mini_m1
    integrations:
      app_store_connect: Codemagic Store Key
    environment:
      ios_signing:
        distribution_type: app_store
        bundle_identifier: app_bundle_id
      vars:
        APP_STORE_APPLE_ID: my_apple_id
      flutter: stable
      xcode: latest
      cocoapods: default
    scripts:
      - name: Clean up and update pods
        script: |
         cd ios
         rm -rf Podfile.lock
         flutter clean
         flutter pub get
         pod install
         pod repo update
      - name: Set up code signing settings on Xcode project
        script: |
          xcode-project use-profiles
      - name: Flutter build ipa and automatic versioning
        script: |
          flutter build ipa --release
            --export-options-plist=/Users/builder/export_options.plist
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - flutter_drive.log
    publishing:
      email:
        recipients:
          - yasirali88805g@gmail.com
        notify:
          success: true
          failure: true
      app_store_connect:
        auth: integration
        submit_to_testflight: true
        submit_to_app_store: false

**Reply by dam-ease** [reply]
Hey @yasirali133314!
I see there is progress from the pod install step.
Please can you try building with `flutter build ipa -v --release --export-options-plist=/Users/builder/export_options.plist` so the full logs can be checked?
I see your last build was cancelled and can't see any error logs on my end.

**Reply by yasirali133314** [reply]
@dam-ease I updated my cocapods from 1.15 to 1.16.2 and tried again, here is the issue.


  sqflite 2.3.3+1 (2.4.0 available)
  sqflite_common 2.5.4 (2.5.4+5 available)
  stack_trace 1.11.1 (1.12.0 available)
  string_scanner 1.2.0 (1.4.0 available)
  synchronized 3.1.0+1 (3.3.0+3 available)
  test_api 0.7.2 (0.7.3 available)
  typed_data 1.3.2 (1.4.0 available)
  url_launcher 6.3.0 (6.3.1 available)
  url_launcher_android 6.3.8 (6.3.14 available)
  url_launcher_linux 3.1.1 (3.2.0 available)
  url_launcher_macos 3.2.0 (3.2.1 available)
  url_launcher_web 2.3.2 (2.3.3 available)
  url_launcher_windows 3.1.2 (3.1.3 available)
  uuid 4.4.2 (4.5.1 available)
  vector_graphics 1.1.11+1 (1.1.13 available)
  vector_graphics_codec 1.1.11+1 (1.1.12 available)
  vector_graphics_compiler 1.1.11+1 (1.1.12 available)
  video_player 2.9.1 (2.9.2 available)
  video_player_android 2.6.0 (2.7.16 available)
  video_player_avfoundation 2.6.1 (2.6.2 available)
  video_player_platform_interface 6.2.2 (6.2.3 available)
  video_player_web 2.3.2 (2.3.3 available)
  vm_service 14.2.5 (14.3.1 available)
  wakelock_plus 1.2.7 (1.2.8 available)
  web 0.5.1 (1.1.0 available)
  win32 5.5.3 (5.8.0 available)
  xdg_directories 1.0.4 (1.1.0 available)
Got dependencies!
1 package is discontinued.
131 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.

Archiving com.app.hmde...
Automatically signing iOS for device deployment using specified development team in Xcode project: 4M5HPXHXS6
Running pod install...                                            134.3s
Running Xcode build...                                          
Xcode archive done.                                         22.2s
Failed to build iOS app
Semantic Issue (Xcode): No member named 'memory_order_relaxed' in 'std::memory_order'; did you mean 'std::memory_order_relaxed'?
/Users/builder/clone/ios/Pods/leveldb-library/util/env_posix.cc:839:33

Semantic Issue (Xcode): No member named 'memory_order_relaxed' in 'std::memory_order'; did you mean 'std::memory_order_relaxed'?
/Users/builder/clone/ios/Pods/leveldb-library/util/env_posix.cc:856:34

Encountered error while archiving for device.

**Reply by dam-ease** [reply]
Hello @yasirali133314!
This new error message seems to be a known issue, I see similar one [here](https://github.com/firebase/flutterfire/issues/13564), can you try with the suggested version in this thread?

**Reply by dam-ease** [reply]
Hello @yasirali133314,
Any update for me on this?If there is anything else I can assist you with, please don't hesitate to let me know.
